src/types/onyx/ReportAction.ts

```diff
@@ -0,0 +1,85 @@
+import OriginalMessage, {Reaction} from './OriginalMessage';
+import * as OnyxCommon from './OnyxCommon';
+
+type Message = {
+    /** The type of the action item fragment. Used to render a corresponding component */
+    type: string;
+
+    /** The text content of the fragment. */
+    text: string;
+
+    /** Used to apply additional styling. Style refers to a predetermined constant and not a class name. e.g. 'normal'
+     * or 'strong'
+     */
+    style?: string;
+
+    /** ID of a report */
+    reportID?: string;
+
+    /** ID of a policy */
+    policyID?: string;
+
+    /** The target of a link fragment e.g. '_blank' */
+    target?: string;
+
+    /** The destination of a link fragment e.g. 'https://www.expensify.com' */
+    href?: string;
+
+    /** An additional avatar url - not the main avatar url but used within a message. */
+    iconUrl?: string;
+
+    /** Fragment edited flag */
+    isEdited: boolean;
+
+    isDeletedParentAction: boolean;
+    whisperedTo: number[];
+    reactions: Reaction[];
+};
+
+type Person = {
+    type?: string;
+    style?: string;
+    text?: string;
+};
+
+type ReportActionBase = {
+    /** The ID of the reportAction. It is the string representation of the a 64-bit integer. */
+    reportActionID?: string;
```

Comment:
I know this was a giant PR, but shouldn't `reportActionID` be required + non-null?

Reply:
@blazejkustra

Reply:
I mimicked the proptypes from [here](https://github.com/software-mansion-labs/expensify-app-fork/blob/7fe7fe5c9e297e500d087637472713c491f222cd/src/pages/home/report/reportActionPropTypes.js#L6). There were some usages of `reportActionID` with a default value, that's probably why it's optional.

The thing is that onyx core data types aren't final and we'll edit these as we continue the migration to Typescript (it wasn't feasible to catch all the usages 😅)